### PR TITLE
Support lowercase hex values

### LINKED ISSUE #1225: Lowercase hex (e.g. #fff) do not convert properly
Cannot use lowercase letters in hex, e.g. `#fff`.

**Error details**
If an exception is raised, it is very important that you provide the full error message.
Otherwise members of the fpdf2 community won't be able to help you with your problem.

**Minimal code**
Please include some minimal Python code reproducing your issue:
```python
from fpdf import FPDF

pdf = FPDF()
pdf.set_text_color("#fff")  # Does not convert
pdf.set_text_color("#FFF")  # Does convert
```
If you don't know how to build a minimal reproducible example, please check this tutorial: https://stackoverflow.com/help/minimal-reproducible-example

**Environment**
Please provide the following information:
* **Operating System**: Linux
* **Python version**: Python 3.10.12
* **`fpdf2` version used**: fpdf2-2.7.9

### PR DESCRIPTION
**Pull request by Makaze**
Convert hex strings to uppercase to support case insensitive hex colors.

Fixes #1225  

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] A mention of the change is present in `CHANGELOG.md`
- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

**Comment by Makaze** [comment]
I don't see any specifics in the docs/ folder about this, so I couldn't find anything to add. The docs/ seems to be higher level. Is docstring enough?

**Comment by Makaze** [comment]
Suggested to reopen for docs + test addition.

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ This can also be enabled programmatically with `warnings.simplefilter('default',
 * [`FPDF.write_html()`](https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.write_html): fixed incoherent indentation of long `<ul>` list entries - _cf._ [issue #1073](https://github.com/py-pdf/fpdf2/issues/1073) - thanks to @lcgeneralprojects
 * default values for `top_margin` and `bottom_margin` in `HTML2FPDF._new_paragraph()` calls are now correctly converted into chosen document units.
 * In [text_columns()](https://py-pdf.github.io/fpdf2/extColumns.html), paragraph top/bottom margins didn't correctly trigger column breaks; [issue #1214](https://github.com/py-pdf/fpdf2/issues/1214)
+* [`fpdf.drawing.color_from_hex_string`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.color_from_hex_string) did not text or mention accepting lowercase hex values.
```
Comment (gmischler) [inline CHANGELOG.md:41]: text -> test

**Review by gmischler — APPROVED** [review]
Just a minor typo left to fix.

Thanks for improving our tests and documentation!

**Comment by gmischler** [comment]
@allcontributors please add @Makaze for documentation, test

**Pull request merged**